File: spring-boots/spring-data-mongo-docker/src/main/java/ru/gulash/spring/service/UserService.java
package ru.gulash.spring.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import ru.gulash.spring.domain.User;
import ru.gulash.spring.repostory.UserRepository;

import java.util.List;

@Service
public class UserService {

    @Autowired
    private UserRepository userRepository;

    public List<User> findUsersByName(String name) {
        return userRepository.findByName(name);
    }

    public List<User> findUsersOlderThan(int age) {
        return userRepository.findByAgeGreaterThan(age);
    }

    public List<User> findUsersByNameAndEmail(String name, String email) {
        return userRepository.findByNameAndEmail(name, email);
    }
}

File: spring-boots/tests/kotlin-mockk-example/src/main/kotlin/com/sprboot/mockkexample/kotlinmockkexample/service/UserService.kt
package com.sprboot.mockkexample.kotlinmockkexample.service

import com.sprboot.mockkexample.kotlinmockkexample.domain.User
import org.springframework.stereotype.Service

@Service
class UserService {
    fun getUserById(id: Long): User {
        // Здесь можно реализовать логику для получения пользователя из базы данных
        return User(id, "User$id")
    }
}